base/linalg/cholesky.jl

```diff
@@ -47,6 +44,7 @@ function CholeskyPivoted{T}(A::AbstractMatrix{T}, uplo::Char, piv::Vector{BlasIn
     CholeskyPivoted{T,typeof(A)}(A, uplo, piv, rank, tol, info)
 end
 
+RealSymOrHerm{S} = Union{Symmetric{<:Real,S}, Hermitian{<:Any,S}}
```

Comment:
[This](https://github.com/JuliaLang/julia/blob/19e3f1bd3df3eb6a00dcbdbb7df6a70ac27704a0/base/linalg/symmetric.jl#L87)? I thought that might not give the same type intersection but I guess it will?